src/Impl/RestPublisher.php

```diff
@@ -47,12 +48,29 @@ private function convertToPayloadTypeName($channelSchemaName): string
 
     public function send(Message $message): SuccessResult
     {
-        $json = $this->jsonProvider->getJson($message, $this->payloadTypeName);
-        $actualHeaders = $message->action == Message::PUBLISH_ACTION
-            ? array_merge($this->headers, ['Content-Type' => 'application/json; charset=UTF-8'])
-            : $this->headers;
+        $messageStatus = $this->sendMulti([$message])[0];
 
-        return $this->httpRequester->executePost($this->messageIngestionEndpointUri, $actualHeaders, $json);
+        if ($messageStatus->getSuccess() != null) {
+            return $messageStatus->getSuccess();
+        }
+
+        $failureResponse = $messageStatus->getFailure();
+        throw StreamxClientExceptionFactory::create(
+            $failureResponse->getErrorCode(),
+            $failureResponse->getErrorMessage()
+        );
```

Comment:
Lines 53-61 moved from Http Requester impl, as specific only for sending a single message

Reply:
BTW.. I think Http Requester should only send the request, and return Http Response.

Now it's a bit of mixing responsibilities. Same in java client:
```java
public interface HttpRequester extends Closeable {

  /**
   * Perform POST request.
   *
   * @param endpointUri Request target URI.
   * @param headers     Request headers
   * @param queryParams Query parameters map.
   * @param jsonWriter  Request entity json content writer.
   * @return Success response entity. <--------------------- this is too specific, should return http response in my opinion. The class name, and the "Perform POST request." javadoc say that it shouldn't do any post-processing of the response
   * @throws StreamxClientException If request failed.
   */
  SuccessResult executePost(
      URI endpointUri,
      Map<String, String> headers,
      Map<String, String> queryParams,
      JsonWriter jsonWriter)
      throws StreamxClientException;
```